modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -754,6 +754,90 @@ func (suite *KeeperTestSuite) TestChanUpgradeOpen() {
 	}
 }
 
+func (suite *KeeperTestSuite) TestChanUpgradeOpenCounterPartyStates() {
+	var path *ibctesting.Path
+	testCases := []struct {
+		name     string
+		malleate func()
+		expError error
+	}{
+		{
+			"success, counterparty in OPEN",
```

Comment:
ya, this is a tradeoff really, we can add these in the main test but require each test case call the ugprade handshake explicitly or we can separate these. Colin also mentioned how he preferred the separation (ref: https://github.com/cosmos/ibc-go/pull/3844#discussion_r1238661755).

I'm open to both options, I think this way it might be slightly cleaner.

Reply:
i would be fine with merging this pr as is :)